modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectDefinitionLocalServiceImpl.java

```diff
@@ -195,11 +195,21 @@ public ObjectDefinition addCustomObjectDefinition(
 			WorkflowConstants.STATUS_DRAFT, objectFields);
 	}
 
-	@Indexable(type = IndexableType.REINDEX)
 	@Override
+	public ObjectDefinition addObjectDefinition(
+		String externalReferenceCode, long userId, long objectFolderId,
+		boolean modifiable, boolean system)
+		throws PortalException {
+		return addObjectDefinition(
+			externalReferenceCode, userId, objectFolderId, modifiable, false,
+			0, system);
+	}
+
+	@Indexable(type = IndexableType.REINDEX)
 	public ObjectDefinition addObjectDefinition(
 			String externalReferenceCode, long userId, long objectFolderId,
-			boolean modifiable, boolean system)
+			 boolean modifiable, boolean rootNode, long rootObjectDefinitionId,
+			boolean system)
```

Comment:
I don't like the idea of adding a parameter in this type of method (add/update) that is not a table property.
I think we should create a validation without the need to pass `boolean rootNode`.

Reply:
we could do like:
if  ID == 0 ; else ....

Reply:
The problem is that we need to know externally for the creation of dummy objects. Maybe the best approach then its to create the dummy object and then do an update?

Reply:
I`ll remove this Overloaded method and pass `rootObjectDefinitionId` in all calls and do an update after the creation of the root empty shell object